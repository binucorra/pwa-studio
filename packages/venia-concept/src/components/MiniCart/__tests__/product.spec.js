import React, { useState } from 'react';
import { createTestInstance } from '@magento/peregrine';

import Product from '../product';

global.getComputedStyle = jest.fn().mockReturnValue({
    getPropertyValue: jest.fn().mockReturnValue('80px')
});
jest.mock('react', () => {
    const React = jest.requireActual('react');
    return Object.assign(React, { useState: jest.fn(React.useState) });
});

jest.mock('../productOptions');
jest.mock('../kebab');

const props = {
    beginEditItem: jest.fn(),
    currencyCode: 'US',
    item: {
        image: {
            file: 'unittest'
        },
        name: 'Unit Test Product',
        options: [],
        price: 99,
        qty: 1
    },
    removeItemFromCart: jest.fn()
};

test('it renders correctly', () => {
    const tree = createTestInstance(<Product {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
});

test('it renders a mask div above the kebab if it is loading', () => {
    useState
        // backgroundImage
        .mockReturnValueOnce([null, jest.fn()])
        // isFavorite
        .mockReturnValueOnce([false, jest.fn()])
        // isLoading
        .mockReturnValueOnce([true, jest.fn()]);

    const tree = createTestInstance(<Product {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
});
