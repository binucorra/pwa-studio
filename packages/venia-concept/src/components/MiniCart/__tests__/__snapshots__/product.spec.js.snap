// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders a mask div above the kebab if it is loading 1`] = `
<li>
  <div
    style={
      Object {
        "backgroundImage": null,
      }
    }
  />
  <div>
    Unit Test Product
  </div>
  ( Product Options Component Here )
  <div>
    <div>
      <span>
        1
      </span>
      <span>
        ×
      </span>
      <span>
        <div />
      </span>
    </div>
  </div>
  <div />
  ( Kebab Component Here )
</li>
`;

exports[`it renders correctly 1`] = `
<li>
  <div
    style={
      Object {
        "backgroundImage": "url(/media/catalog/product/unittest?auto=webp&format=pjpg&width=80)",
      }
    }
  />
  <div>
    Unit Test Product
  </div>
  ( Product Options Component Here )
  <div>
    <div>
      <span>
        1
      </span>
      <span>
        ×
      </span>
      <span>
        <div />
      </span>
    </div>
  </div>
  ( Kebab Component Here )
</li>
`;
